### PR TITLE
feat(rule-finder): add --ext option

### LINKED ISSUE #322: Handle typescript-eslint configuration
When it runs in a project containing only "*.ts" files (so not any "*.js" file) this error occurs:
image

It seems that the issue come from this line : https://github.com/sarbbottam/eslint-find-rules/blob/ab323d6746ec9f60d24d0a26929e4ea01275d17a/src/lib/rule-finder.js#L84

It could be nice to have a way to configure the glob pattern or maybe just the extension part (see --ext option in eslint cli https://eslint.org/docs/user-guide/command-line-interface#options)

Thanks!

### PR DESCRIPTION
**Pull request by nicolashenry**
fix #322 
~depends on #324~

### CODE REVIEW

File: src/lib/rule-finder.js [hunk 1]
```diff
@@ -78,10 +78,9 @@ function _isNotCore(rule) {
   return rule.indexOf('/') !== '-1';
 }
 
-function RuleFinder(specifiedFile, options) {
-  const {omitCore, includeDeprecated} = options;
+function RuleFinder(specifiedFile, {omitCore, includeDeprecated, ext = ['.js']}) {
   const configFile = _getConfigFile(specifiedFile);
-  const files = glob.sync('**/*.js', {dot: true, matchBase: true});
+  const files = glob.sync(`**/*@(${ ext.join('|') })`, {dot: true, matchBase: true});
```
Comment (ljharb) [inline src/lib/rule-finder.js:83]: can we ensure that the extensions don't contain `*`, or `./` or `../`? also, does eslint allow an ext like `.a.b`, or must it be a single one?

Comment (nicolashenry) [reply src/lib/rule-finder.js:83]: I will do a post filter instead then. I think that eslint allow multi dots extensions but I am not 100% sure.

**Comment by nicolashenry** [comment]
@ljharb I aligned this PR with what eslint does : https://github.com/eslint/eslint/blob/3a5fbb3d634be688615950c0a5fa8aead6ff08b5/lib/cli-engine/file-enumerator.js#L150

Note that I had to drop unicode flag in regexp compared to eslint because of node 4 support in tests (I think you should drop support for node 4 and 6, it's pretty old now).

**Comment by nicolashenry** [comment]
@ljharb Should I change something else?

File: src/lib/rule-finder.js [hunk 2]
```diff
@@ -78,10 +78,27 @@ function _isNotCore(rule) {
...
+function _escapeRegExp(str) {
+  return str.replace(/[\\^$.*+?()[\]{}|]/g, '\\$&');
+}
+
+function _createExtensionRegExp(extensions) {
+  const normalizedExts = extensions.map(ext => _escapeRegExp(
+    ext.startsWith(".")
+      ? ext.slice(1)
+      : ext
+  ));
```
Comment (ljharb) [inline src/lib/rule-finder.js:90]:
```suggestion
  const normalizedExts = extensions.map(ext => _escapeRegExp(
    ext.startsWith('.') ? ext.slice(1) : ext
  ));
```

**Comment by nicolashenry** [comment]
@ljharb It would be great to have a new release :) (when you have some time)

**Comment by ljharb** [comment]
#327

**Pull request merged**